### PR TITLE
add rpc warning

### PR DESCRIPTION
**Pull request by zielvna**

### CODE REVIEW

File: src/store/sagas/index.ts [hunk 1]
```diff
@@ -7,6 +7,21 @@ import { positionsSaga } from '@sagas/positions'
 import { statsHandler } from './stats'
 import { farmsSaga } from './farms'
 import { bondsSaga } from './bonds'
+import { RpcStatus, actions as solanaConnectionActions } from '@reducers/solanaConnection'
+import { put, select } from 'typed-redux-saga'
+import { rpcStatus } from '@selectors/solanaConnection'
+
+export function* handleError(error: Error): Generator {
```
Comment (wojciech-cichocki) [inline src/store/sagas/index.ts:14]: I think we could change the name of this to e.g. `handleRpcError` ro sth similar

File: src/store/sagas/positions.ts [hunk 2]
```diff
@@ -760,6 +765,8 @@ export function* handleInitPosition(action: PayloadAction<InitPositionData>): Ge
         })
       )
     }
+
+    yield* call(handleError, error as Error)
```
Comment (wojciech-cichocki) [inline src/store/sagas/positions.ts:769]: Is it possible for this function not to be repeated every saga?

**Review by wojciech-cichocki — CHANGES_REQUESTED** [review]

**Review by p6te — COMMENTED** [review]
you can add handle error also for connect wallet, for example for rpc quicknode, user can't event connect wallet

**Comment by p6te** [comment]
> you can add handle error also for connect wallet, for example for rpc quicknode, user can't event connect wallet

and also for swap simulation, because it can not work, and user doesn't know why

**Comment by p6te** [comment]
there is more error cases which are for Alchemy RPC, so errors can be different. 
- close position
image
- reload chart(handleGetCurrentPlotTicks saga)
image

you can trigger handleError for errors: 400 and 429 too

File: src/store/sagas/positions.ts (hunk 2 above)
Comment (zielvna) [reply src/store/sagas/positions.ts:769]: maybe global try catch on all sagas would work but we would have to rethrow that error every try catch in each saga anyway

**Comment by p6te** [comment]
We can avoid crashing the app when user decided to use not recommended RPC. The case is when you choose any other RPC than Helius or Alchemy, write some number in the first input of swap, and then try to write something in the second input. Even if rpc is not working correctly we should avoid crashing the app.

**Review by p6te — APPROVED** [review]

**Review by wojciech-cichocki — APPROVED** [review]

**Pull request merged**